Sources/StreamChat/Utils/Database/NSManagedObject+Extensions.swift

```diff
@@ -16,6 +16,7 @@ extension NSEntityDescription {
         for request: NSFetchRequest<T>
     ) -> T {
         let entity = insertNewObject(forEntityName: T.entityName, into: context)
+        request.entity = NSEntityDescription.entity(forEntityName: T.entityName, in: context)!
```

Comment:
Why is this needed? Aren't we doing that already? Also, seems unsafe

Reply:
Casting `request` to `NSFetchRequest<NSFetchRequestResult>` requires `request` to have an `entity` set, I've seen it crashing because of this. It's safe since `T.entityName` is guaranteed to be a registered entity name

Reply:
let's do an if let there, with an assertion failure in the else. To play it safe. WDYT?

Reply:
You should be able to write these two lines as follows and not have to worry about the entity name:

```
let entity = T(context: context)
```

That should work as long as the entity you're initializing is correctly set up in the model editor where the model name is connected to the correct class name (which is should be because you'd see various issues otherwise).

Reply:
fixed!